### PR TITLE
Add the Disguise Self spell

### PR DESCRIPTION
**Pull request by datlo**
## What Does This PR Do

Adds the Disguise Self spell to the wizard's spellbook for 1 point. 10 second cooldown.
Lets you change your appearance and voice, working similarly to the abductor's disguise.
Examining the disguised wizard directly shows their real appearance and worn items.
Breaks on getting damaged/shot by disablers. 
The disguise is picked from a random crewmember with access to the room you're in if possible. 
Recasting the spell when disguised rolls a new disguise.
Replace the Summon Item spell with this one for stealth wizard apprentice.

## Why It's Good For The Game

Fun addition to the wizard's kit. The disguise status effect is also planned to be used by the Mindflayer antag.
Captain and HOP can't be rolled as disguises as they are too remarkable and have too much access.
Clown can't be rolled as disguise because waddle and squeak are noticeably absent from the disguise.

## Testing

Tested many edge cases by parsing all mobs instead of just players, readded the player check and did basic testing using 2 accounts to check that everything works as intended

## Changelog
:cl:
add: Added the Disguise Self spell to the wizard spellbook for 1 point. Spell icon sprite by McRamon
tweak: Summon item has been replaced by Disguise Self for stealth apprentices.
/:cl:

### CODE REVIEW

File: code/datums/status_effects/magic_disguise.dm [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		if((ACCESS_CAPTAIN in disguise_source.get_access()) || (ACCESS_HOP in disguise_source.get_access()) || (ACCESS_CLOWN in disguise_source.get_access()))
+			continue //We don't want the cap, HOP or clown as a disguise, too remarkable. If you're spotted by the Cap or HOP in their own office, disguising as them wont help you either
+		if((!AL || AL.allowed(disguise_source)) && !disguise_source.mind.offstation_role && disguise_source != owner)
+			return disguise_source
+	for(var/mob/living/carbon/human/backup_source in shuffle(GLOB.player_list)) //Pick a random crewmember if there's no one with access to the current room
+		if((ACCESS_CAPTAIN in backup_source.get_access()) || (ACCESS_HOP in backup_source.get_access()) || (ACCESS_CLOWN in backup_source.get_access()))
+			continue //ditto
+		if(!backup_source.mind.offstation_role && backup_source != owner)
+			return backup_source
+	return null
```
Comment (DGamerL) [inline code/datums/status_effects/magic_disguise.dm:60]:
```suggestion
	return
```

File: code/datums/status_effects/magic_disguise.dm [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+/datum/status_effect/magic_disguise/proc/create_disguise(mob/living/carbon/human/disguise_source)
+	var/datum/icon_snapshot/temp = new
+	temp.name = disguise_source.name
+	temp.icon = disguise_source.icon
+	temp.icon_state = disguise_source.icon_state
+	temp.overlays = disguise_source.get_overlays_copy(list(L_HAND_LAYER,R_HAND_LAYER))
```
Comment (DGamerL) [inline code/datums/status_effects/magic_disguise.dm:67]:
```suggestion
	temp.overlays = disguise_source.get_overlays_copy(list(L_HAND_LAYER, R_HAND_LAYER))
```

**Comment by McRamon** [comment]
image
please use this icon
contact me on discord for dmi

**Comment by datlo** [comment]
Added icon, thank you

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**